core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -41,8 +41,11 @@ class NewShinyConsumer(topic: String, consumerProps: Properties) extends BaseCon
   var recordIter = consumer.poll(0).iterator
 
   override def receive(): BaseConsumerRecord = {
-    while (!recordIter.hasNext)
-      recordIter = consumer.poll(Long.MaxValue).iterator
+    while (!recordIter.hasNext) {
+      recordIter = consumer.poll(timeoutMs).iterator
+      if (!recordIter.hasNext)
+        throw new ConsumerTimeoutException
+    }
```

Comment:
It seems like the `while` loop is no longer needed given the new `if` condition that is exactly the same as the `while` condition, but breaks out of the loop via an exception. Right?